ton_client/client/src/tvm/run_message.rs

```diff
@@ -27,21 +27,98 @@ use super::types::{ExecutionOptions, ResolvedExecutionOptions};
-use ton_block::{Message, Serializable};
+use ton_block::{Account, Message, Serializable};
 use ton_executor::{ExecutorError, OrdinaryTransactionExecutor, TransactionExecutor};
 use ton_sdk::TransactionFees;
 use ton_types::Cell;
 
+#[derive(Serialize, Deserialize, ApiType, Debug, Clone)]
+#[serde(tag="type")]
+pub enum AccountForExecutor {
+    /// Non-existing account to run a creation internal message.
+    /// Should be used with `skip_transaction_check = true` if the message has no deploy data
+    /// since transaction on the unitialized account are always aborted
+    None,
+    /// Emulate unitialized account to run deploy message
+    Uninit,
+    /// Account state to run message
+    State {
+        /// Account BOC. Encoded as base64.
+        boc: String,
+        /// Flag for running account with the unlimited balance. Can be used to calculate
+        /// transaction fees without balance check
+        unlimited_balance: Option<bool>,
+    },
+}
+
+const UNLIMITED_BALANCE: u64 = std::u64::MAX;
+
+impl AccountForExecutor {
+    pub fn get_account(
+        &self,
+        context: &Arc<ClientContext>,
+        address: ton_block::MsgAddressInt
+    ) -> ClientResult<(Cell, Option<ton_block::CurrencyCollection>)> {
+        match self {
+            AccountForExecutor::None => {
+                let account = Account::AccountNone
+                    .write_to_new_cell()
+                    .unwrap()
+                    .into();
+                Ok((account, None))
+            },
+            AccountForExecutor::Uninit => {
+                let last_paid = (context.env.now_ms() / 1000) as u32;
+                let account = Account::uninit(address, 0, last_paid, UNLIMITED_BALANCE.into());
+                let account = serialize_object_to_cell(&account, "account")?;
+                Ok((account, None))
+            },
+            AccountForExecutor::State { boc, unlimited_balance } => {
+                if unlimited_balance.unwrap_or_default() {
+                    let mut account: Account = deserialize_object_from_base64(&boc, "account")?
+                        .object;
+                    let original_balance = account.get_balance()
+                        .ok_or(Error::invalid_account_boc("can not set unlimited balance for non existed account"))?
+                        .clone();
+                    let mut balance = original_balance.clone();
+                    balance.grams = UNLIMITED_BALANCE.into();
+                    account.set_balance(balance);
+                    let account = serialize_object_to_cell(&account, "account")?;
+                    Ok((account, Some(original_balance)))
+                } else {
+                    let (_, account) = deserialize_cell_from_base64(&boc, "account")?;
+                    Ok((account, None))
+                }
+            }
+        }
+    }
+
+    pub fn restore_balance(
+        account: Cell,
+        balance: Option<ton_block::CurrencyCollection>
```

Comment:
Suggest to remove Option here. Caller must provide balance or must not call this function.

Reply:
Options is here to keep caller code clear. Renamed to `restore_balance_if_needed`